service/src/main/java/bio/terra/externalcreds/PublicApiController.java

```diff
@@ -0,0 +1,35 @@
+package bio.terra.externalcreds;
+
+import bio.terra.externalcreds.generated.api.PublicApi;
+import java.sql.SQLException;
+import java.util.Objects;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@Controller
+public class PublicApiController implements PublicApi {
+
+  final JdbcTemplate jdbcTemplate;
+
+  public PublicApiController(JdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  @Override
+  @GetMapping("/status")
+  public ResponseEntity<Void> getStatus() {
+    try {
+      if (Objects.requireNonNull(this.jdbcTemplate.getDataSource()).getConnection().isValid(0)) {
+        return new ResponseEntity<>(HttpStatus.OK);
+      } else {
+        return new ResponseEntity<>(HttpStatus.EXPECTATION_FAILED);
```

Comment:
This should not be a 4xx level error. 4xx errors generally means the client did something wrong and can fix it. I think this should be INTERNAL_SERVER_ERROR. If we really want to distinguish this case from the one below I would suggest 503 but 503 is usually indicative of an apache proxy error so I don't recommend it. The real thing that we should use to distinguish the error condition is logging. Add some logging :)